lib/Util/Notices/DiscountsAnnouncement.php

```diff
@@ -0,0 +1,44 @@
+<?php
+namespace MailPoet\Util\Notices;
+
+use MailPoet\Models\Setting;
+use MailPoet\Util\Helpers;
+use MailPoet\WP\Notice as WPNotice;
+
+class DiscountsAnnouncement {
+
+  const OPTION_NAME = 'mailpoet_display_discounts_announcement';
+
+  function enable() {
+    Setting::setValue(self::OPTION_NAME, true);
+  }
+
+  function disable() {
+    Setting::setValue(self::OPTION_NAME, false);
+  }
+
+  function init($should_display) {
+    $should_display = $should_display && (time() <= strtotime('2018-11-30 23:59:59'));
+    if($should_display && Setting::getValue(self::OPTION_NAME, true)) {
+      return $this->display();
+    }
+  }
+
+  private function display() {
+    $message = Helpers::replaceLinkTags(
+      __('<h3>Save on MailPoet Premium for a limited time. Discounts up to 40%</h3>
+          <p>Our annual sale is a good opportunity to get more detailed stats & great email deliverability. Don\'t miss out!</p>
```

Comment:
You can use `’` character instead of `'` in `Don\'t`. You will be able to remove escaping.